layers/+tools/shell/funcs.el

```diff
@@ -174,6 +174,20 @@ is achieved by adding the relevant text properties."
   (define-key eshell-mode-map
     (kbd "M-l") 'spacemacs/helm-eshell-history))
 
+(defun spacemacs/ivy-eshell-history ()
+  (interactive)
+  (counsel-esh-history)
+  ;; (evil-insert-state)
+  )
+
+(defun spacemacs/init-ivy-eshell ()
+  "Initialize ivy-eshell."
+  (spacemacs/set-leader-keys-for-major-mode 'eshell-mode
+    "H" #'spacemacs/ivy-eshell-history)
+  (define-key eshell-mode-map (kbd "M-l") #'spacemacs/ivy-eshell-history)
+  (define-key eshell-mode-map (kbd "<tab>")
+    (lambda () (interactive) (pcomplete-std-complete))))
```

Comment:
Maybe this should be a named function, like `spacemacs/ivy-eshell-history`. 

At least one case where a named function would be helpful is in the ` C-h l` (lower case L) (`view-lossage`) buffer.

With a lamda the lossage entry looks like this:
> \<tab> [anonymous-command]

With a named function, for example like this:
```
(defun spacemacs/pcomplete-std-complete ()
  (interactive)
  (pcomplete-std-complete))
```

And in `spacemacs/init-ivy-eshell`
```
  (define-key eshell-mode-map (kbd "<tab>") #'spacemacs/pcomplete-std-complete)
```

Then the entry is more descriptive:
> \<tab> [spacemacs/pcomplete-std-complete]

Reply:
Definitely. I rarely use `view-lossage` and often overlook what "lossage" shows.